src/System.IO.Pipelines/PipeFactory.cs

```diff
@@ -99,7 +113,11 @@ public IPipeWriter CreateWriter(IPipeWriter writer, Func<IPipeReader, IPipeWrite
         public IPipeReader CreateReader(IPipeReader reader, Func<IPipeReader, IPipeWriter, Task> produce)
         {
             var pipe = new Pipe(_pool);
-            produce(reader, pipe);
+
+            produce(reader, pipe).ContinueWith(t =>
+            {
+            });
```

Comment:
This isn't observing anything. Not only in the sense it is ignored, but also because the Task.Exception property is never accessed. This will still raise an UnobservedTaskException just like before.

Once again, I recommend calling this in an async method and using try/catch. Use `var ignore =` or a pragma to disable warnings if need be. Even async void is defensible if completely wrapped by a try/catch.